### ISSUE TITLE
DAG Run Grid View: Run Id not updated consistently

### ISSUE THREAD

**Issue by AchimGaedkeLynker** [apache/airflow#36320]
### Apache Airflow version

2.8.0

### What happened?

I use the Run ID quite often to refer to the results of DAG runs, so I rely on the copy function (the two squares logo on the right of the run id text). Though the Run ID text next to the copy button is updated consistently when clicking into the grid, the string copied to the clipboard is not (i.e. it is different than the text displayed). - This actually leads to confusion.

![image](https://github.com/apache/airflow/assets/135793393/ac0ea8d4-083f-4684-aa60-d91f4f39309e)

### What you think should happen instead?

When clicking at any square of the grid (i.e not the top row bars reflecting the duration of the DAG run), the text in the dag task instance details updates and so should the text sent to the clipboard.

### How to reproduce

Take any of your DAGs, click inside the grid on a square and observe the Run Id text changing.
In parallel to this, use the "copy" button at the right side of the text and realize that it is constant/outdated.

The text and the copy text change consistently when clicking on the bars on the top row.

I observed this with Chrome and Firefox.

### Operating System

Microsoft Windows

### Deployment

Virtualenv installation

### Deployment details

Using ssh port-forwarding to access the web server.

### Anything else?

We've experienced that also at version 2.7.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
Duplicate of:
- https://github.com/apache/airflow/issues/34846
